### ISSUE TITLE
Functionality broken with ef release 6.2?

### ISSUE THREAD

**Issue by srudin** [moozzyk/EFInteractiveViews#7]
I have tried to use the package according to this blog post: https://www.fusonic.net/en/blog/ef-cache-deployment/ 

I also adopted the edmx loading which is now integrated in ef 6.2. 

Both the edmx loading and the view cache loading are working - when I activate them seperately. When I do both I get the following error: 

The entity type EdmMetadata is not part of the model for the current context. 

on calling InteractiveViews.SetViewCacheFactory(this, cacheFactory); 

I have the impression that the 6.2 release has changed something that breaks this package...?

**Comment by srudin** [comment]
I have now changed the logic to pass a DbCompiledModel to the constructor of the context (instead of using the DefaultModelStore as shown in the blog post I linked above).

This has changed the behaviour: The SetViewCacheFactory call now works. BUT instead the Create call (to initially save the views) now throws with:

Unable to calculate model hash because no mapping was found between conceptual model container 'AuditEnabledUnitOfWorkContext' and store model container 'CodeFirstDatabase'. Ensure that the names match those defined in the EDMX or Code First model.

Is something broken or am I doing something wrong?

**Comment by moozzyk** [comment]
I haven't tried EF6.2 yet. Can you post a simple repro?

**Comment by srudin** [comment]
I have been able to get it to work - well, partially.

It works when passing the DbCompiledModel to the constructor; it doesn't work using the DefaultDbModelStore. This looks more like an EntityFramework problem so I posted it there: https://github.com/aspnet/EntityFramework6/issues/275

I started with the first approach then changed it to the latter and didn't realized this was the problem and not the combination as I stated in my first post - sorry for that. I still believe that the solution from the blog post is NOT working with EF6.2 but it seems to be because of EF and not your package. I'll keep you posted when I know more.

**Comment by moozzyk** [comment]
The blog post was written long before EF 6.2 shipped. It seems that EF 6.2 has a new feature that at least partially does not work with the EFInteractiveViews. I need to figure out if there is a way to make this package work with EF6.2 (without breaking previous releases) or the if the issue is that the new feature was not tested with views and therefore is not working. Having a simple repro would make it much easier for me to investigate.

**Comment by srudin** [comment]
Glad to hear you agree that there has been some change...
Unfortunately the solution I'm working on is a pretty large enterprise project. I cannot share it with you and even if I could it wouldn't be a good ground to test on. Having said that maybe also explains why it was/is not to easy for me to say what exactly the problem is ;-)

**Comment by moozzyk** [comment]
I was hoping you could create a simplified repro (e.g. one and a very simple table) based on your findings...

**Comment by srudin** [comment]
Following one of the answers in my post in the ef repo I used the DefaultModelStore without changing the file path. This works indeed. I have no idea what the difference is (the generated files are binary identical) but I don't have the time to search in the source code so be it this way.

Basically this means that everything is working - but NOT the way it is described in the blog post I mentioned above and which I was following at the beginning (https://www.fusonic.net/en/blog/ef-cache-deployment/). The differences are:

1. Do not generate the Edmx yourself.
2. In your DbConfiguration use SetModel with the standard DefaultModelStore only.
3. Do not call SetViewCacheFactory in the context constructor. Instead instantiate a new context during application startup (it has to be the first instantiation), make the call and dispose the context.

This works for me and I can live with that so no further action required. Sorry for the issue - basically there was no problem with this package, only a change of behaviour compared to the versions used in the blog post.

**Comment by moozzyk** [comment]
I think I pointed out in the [post](If you look at the [post](https://blog.3d-logic.com/2013/12/14/using-pre-generated-views-without-having-to-pre-generate-views-ef6/) I wrote a while ago to announce EFInteractiveViews you will notice that I recommended setting the view cache factory ) announcing interactive views the that initialization has to happen before performing any operation that involves DbContext - e.g. in the **static** constructor.
Having said that - have you ever measured whether interactive views help at all? I don't know how big your model is but if you have less than 50 entities you will likely not notice any difference and hence interactive views just add complexity without giving any benefits. I would recommend measuring application start-up improvement when using interactive views and only use interactive views if they actually help.

**Comment by srudin** [comment]
Well as I said I found EFInteractiveViews only because I followed the posts from fusonic. I followed their guide and as you can see they set it in the (non-static) constructor so that's what I tried at first.
Sure I tested the impact - it's about 5-10s. But we have 5 databases.
